strawberryfields/apps/similarity.py

```diff
@@ -304,7 +304,13 @@ def orbit_cardinality(orbit: list, modes: int) -> int:
     """
     sample = orbit + [0] * (modes - len(orbit))
     counts = list(Counter(sample).values())
-    return int(factorial(modes) / np.prod(factorial(counts)))
+
+    # factorials of numbers larger than 170 do not fit into a int,
+    # hence return float using the qarg `exact=True`
+    if modes > 170:
+        return factorial(modes, exact=True) / np.prod(factorial(counts, exact=True))

```

Comment:
@sduquemesa Would be good to add a test for this!

How come `exact=True` isn't used everywhere instead of having this if-statement? Is this due to it being much slower, or some other reason? 🤔

Reply:
@thisac, this is explained on scipy factorila's [documentation](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.factorial.html)
> If True, calculate the answer exactly using long integer arithmetic. If False, result is approximated in floating point rapidly using the gamma function. Default is False.

Just added a test that will cover this case! 🩺